### PR TITLE
Feature/167 calculate hourly tariff flex mp

### PR DESCRIPTION
**Pull request by ZavezX**
## References

* #0000

### CODE REVIEW

File: source/coordinator/GreenEnergyHub.Aggregation.Application/Coordinator/CoordinatorService.cs [hunk 1]
```diff
@@ -189,5 +123,91 @@ public async Task HandleResultAsync(string inputPath, string resultId, string pr
 
             _logger.LogInformation("Message handled {inputPath} {resultId} {processType} {startTime} {endTime}", inputPath, resultId, processType, startTime, endTime);
         }
+
+        private async Task CreateAndRunDatabricksJobAsync(string processType, string jobName, List<string> parameters, string pythonFileName, CancellationToken cancellationToken)
```
Comment (jonasdmoeller) [inline source/coordinator/GreenEnergyHub.Aggregation.Application/Coordinator/CoordinatorService.cs:127]: I think this method should be refactored. One method responsible for the entire databricks cluster startup is not pretty. Especially the thread sleep, but I dont know if those can be avoided. But it is not part of this PR, so it wont prevent approval.

File: source/coordinator/GreenEnergyHub.Aggregation.Application/Coordinator/CoordinatorService.cs [hunk 2]
```diff
@@ -189,5 +123,91 @@ public async Task HandleResultAsync(string inputPath, string resultId, string pr
...
+        {
+            // TODO: #199 beginTime, endTime, resultId and persist should stored when creating job metadata (add to parameters list)
```
Comment (jonasdmoeller) [inline source/coordinator/GreenEnergyHub.Aggregation.Application/Coordinator/CoordinatorService.cs:129]: Having TODO comments in production code is a code smell. If necessary they should link to some github/zenhub issue.

File: source/coordinator/GreenEnergyHub.Aggregation.Coordinator/CoordinationTriggers.cs [hunk 3]
```diff
@@ -122,24 +135,120 @@ public CoordinationTriggers(ICoordinatorService coordinatorService)
...
+            if (!bool.TryParse(req.Query["persist"], out var persist))
+            {
+                throw new ArgumentException($"Could not parse value {nameof(persist)}");
+            }
+
+            // Because this call does not need to be awaited, execution of the current method
+            // continues and we can return the result to the caller immediately
+            #pragma warning disable CS4014
+
+            // TODO: #199 add processVariant to StartWholesaleJobAsync and store it as metadata
```
Comment (jonasdmoeller) [inline source/coordinator/GreenEnergyHub.Aggregation.Coordinator/CoordinationTriggers.cs:247]: Link to zenhub issue

File: source/coordinator/GreenEnergyHub.Aggregation.Domain/DTOs/MetaData/Job.cs [hunk 4]
```diff
@@ -28,6 +28,7 @@ public Job(string processType)
             Created = SystemClock.Instance.GetCurrentInstant();
             ProcessType = processType;
             State = "Job created";
+            Owner = "Owner"; // TODO: #199 Fill out owner
```
Comment (jonasdmoeller) [inline source/coordinator/GreenEnergyHub.Aggregation.Domain/DTOs/MetaData/Job.cs:31]: Link to zenhub issue

File: source/databricks/geh_stream/wholesale_utils/wholesale_initializer.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+from pyspark.sql.dataframe import DataFrame
+from pyspark.sql.functions import col
+from geh_stream.schemas import time_series_schema, metering_point_schema, grid_loss_sys_corr_schema, market_roles_schema, charges_schema, charge_links_schema, charge_prices_schema, es_brp_relations_schema
+import dateutil.parser
+
+
+def get_hourly_charges(charges: DataFrame, charge_links: DataFrame, charge_prices: DataFrame) -> DataFrame:
+    hourly_charges = charges \
+        .filter(col("resolution") == "PT1H") \
```
Comment (kristianschneider) [inline source/databricks/geh_stream/wholesale_utils/wholesale_initializer.py:23]: Should this not be an argument ? PT1H?

**Review by kristianschneider — COMMENTED** [review]
Looks fine. I have one comment

**Comment by ZavezX** [comment]
@johevemi @kristianschneider @jonasdmoeller Why was this `DRAFT` pull request reviewed and merged?
